apps/OboeTester/app/src/main/java/com/mobileer/oboetester/StreamConfigurationView.java

```diff
@@ -418,4 +486,55 @@ private void onChannelMaskSpinnerSelected() {
         mIsChannelMaskLastSelected = true;
     }
 
+    private void onRequestAudioEffectClicked(boolean isChecked) {
+        if (isChecked){
+            if (misOutput) {
+                mOutputEffectsLayout.setVisibility(VISIBLE);
+            } else {
+                mInputEffectsLayout.setVisibility(VISIBLE);
+            }
+        } else {
+            if (misOutput) {
+                mOutputEffectsLayout.setVisibility(GONE);
+            } else {
+                mInputEffectsLayout.setVisibility(GONE);
+            }
+        }
+    }
+
+    public void setupEffects(int sessionId) {
+        if (misOutput) {
+            mBassBoost = new BassBoost(0, sessionId);
+            mBassBoost.setStrength((short) mBassBoostSeekBar.getProgress());
+            mLoudnessEnhancer = new LoudnessEnhancer(sessionId);
+            mLoudnessEnhancer.setTargetGain((short) mLoudnessEnhancerSeekBar.getProgress());
+        } else {
+            mAcousticEchoCanceler = AcousticEchoCanceler.create(sessionId);
+            if (mAcousticEchoCanceler != null) {
```

Comment:
https://developer.android.com/reference/android/media/audiofx/AcousticEchoCanceler#isAvailable()
might be useful here.

Reply:
Thanks! Added in the latest iteration